### PR TITLE
rsl: add tests for unused GetLatestReferenceUpdaterEntry option combinations and options

### PR DESCRIPTION
**Pull request by aryarathoree**
## Description
This PR adds test coverage for previously unused option combinations in `GetLatestReferenceUpdaterEntry`.

Added coverage for:
- Combined filters:
  - `ForNonGittufReference` + `ForReference`
  - `IsReferenceEntry` + `ForReference`
  - `IsUnskipped` + `ForNonGittufReference`
- The `UntilEntryID` option
- Valid usage of `BeforeEntryNumber`

These tests cover the combinations and options I was able to identify and implement, helping improve coverage of option interactions and previously untested edge cases. All existing tests were passing prior to this change, and this PR does not modify existing behavior.

---

## AI Usage

- [ ] I **did not** use generative AI at all in making the content of this pull
  request.
- [x] I **did** use generative AI in some form in making the content of this
  pull request. I have described my use of AI below.

I used generative AI to help locate relevant functions in the codebase, understand the structure of the repository, and assist in drafting explanations for this pull request. All code changes, test cases, and final validations were written and verified by me.
---
## Contributor Checklist

- [x] I **have manually reviewed all content** submitted to gittuf in this pull
  request.
- [x] I fully understand the content I am submitting.
- [x] The changes introduced are documented and have tests included if
  applicable.
- [x] My changes do not infringe on copyright/trademarks/etc.
- [x] All commits in this pull request include a [DCO
  Signoff](https://wiki.linuxfoundation.org/dco).
- [x] By submitting this pull request, I agree to follow the gittuf [Code of
  Conduct](https://github.com/gittuf/community/blob/main/CODE-OF-CONDUCT.md).

### CODE REVIEW

File: internal/rsl/rsl_test.go [hunk 1]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
 		assert.ErrorIs(t, err, ErrInvalidGetLatestReferenceUpdaterEntryOptions)
 		assert.Nil(t, entry)
 	})
+
+	t.Run("with non gittuf and reference filter", func(t *testing.T) {
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please merge this test with the one on line 651. (`with non gittuf option, mix of gittuf and non gittuf entries`)

File: internal/rsl/rsl_test.go [hunk 2]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(
+			repo,
+			ForNonGittufReference(),
+			ForReference("refs/heads/main"),
+		)
+
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		assert.NotNil(t, entry)
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: The test needs to follow the format of the other tests; instead of `assert.NotNil(t, entry)`, instead use `assert.Equal` and then compare the IDs for example.

Please also make this change for all cases where you've added an `assert.NotNil`.

File: internal/rsl/rsl_test.go [hunk 3]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+			repo,
+			UntilEntryID(firstEntry.GetID()),
+		)
+
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		assert.NotNil(t, entry)
+	})
+
+	t.Run("with before entry number", func(t *testing.T) {
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please move this test above the one on line 718. (`transitioning from no numbers to numbers`)

File: internal/rsl/rsl_test.go [hunk 4]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+		}
+
+		if err := NewReferenceEntry("refs/heads/main", gitinterface.ZeroHash).Commit(repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(
+			repo,
+			BeforeEntryNumber(2),
+		)
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]:
```suggestion
		entry, annotations, err := GetLatestReferenceUpdaterEntry(repo, BeforeEntryNumber(2))
```

Please also make this change every time you call `GetLatestReferenceUpdaterEntry`.

File: internal/rsl/rsl_test.go [hunk 5]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+		}
+
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(
+			repo,
+			ForNonGittufReference(),
+			ForReference("refs/heads/main"),
+		)
+
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: These tests don't use annotations, so you can remove the assert here and just use `_` in the function call for the `annotations` variable.

File: internal/rsl/rsl_test.go [hunk 6]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+			IsUnskipped(),
+			ForNonGittufReference(),
+		)
+
+		assert.Nil(t, err)
+		assert.Empty(t, annotations)
+		assert.NotNil(t, entry)
+	})
+
+	t.Run("with until entry ID", func(t *testing.T) {
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please move this test above the one on line 216. (`with ref name and until entry number`)

File: internal/rsl/rsl_test.go (hunk 1 above)
Comment (patzielinski) [inline internal/rsl/rsl_test.go]:
```suggestion
	t.Run("with ref and non gittuf option", func(t *testing.T) {
```

File: internal/rsl/rsl_test.go [hunk 7]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+			IsReferenceEntry(),
+			ForReference("refs/heads/main"),
+		)
+
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		assert.NotNil(t, entry)
+	})
+
+	t.Run("with unskipped and non gittuf", func(t *testing.T) {
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please move this test above the one on line 651. (`with non gittuf option, mix of gittuf and non gittuf entries` )

Comment (patzielinski) [inline internal/rsl/rsl_test.go]:
```suggestion
	t.Run("with unskipped and non gittuf option", func(t *testing.T) {
```

File: internal/rsl/rsl_test.go [hunk 8]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+			ForNonGittufReference(),
+			ForReference("refs/heads/main"),
+		)
+
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		assert.NotNil(t, entry)
+	})
+
+	t.Run("with reference entry and ref filter", func(t *testing.T) {
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please merge this test into the first test, on line 138. (`with ref name`)

File: internal/rsl/rsl_test.go [hunk 9]
```diff
@@ -877,6 +877,117 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+		if err := NewReferenceEntry("refs/heads/main", gitinterface.ZeroHash).Commit(repo, false); err != nil {
+			t.Fatal(err)
+		}
+
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(
+			repo,
+			ForNonGittufReference(),
+			ForReference("refs/heads/main"),
+		)
+
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]:
```suggestion
```

I suggest to keep the call to `GetLatestReferenceUpdaterEntry` and the following asserts in one block to make it slightly easier to assess which call is being tested.

File: internal/rsl/rsl_test.go [hunk 10]
```diff
@@ -237,6 +263,12 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
 		assert.Nil(t, annotations)
 		assert.Equal(t, rslRef, entry.GetID())
 
+		// reference entry + ref filter
+		entry, annotations, err = GetLatestReferenceUpdaterEntry(repo, IsReferenceEntry(), ForReference(refName))
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		assert.Equal(t, rslRef, entry.GetID())
+
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go]: Please move into the test on line 138 (`with ref name`).

File: internal/rsl/rsl_test.go [hunk 11]
```diff
@@ -213,6 +213,32 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(repo, UntilEntryID(firstEntry.GetID()))
+		assert.Nil(t, err)
+		assert.Nil(t, annotations)
+		latestEntry, err := GetLatestEntry(repo)
+		if err != nil {
+			t.Fatal(err)
+		}
+		assert.Equal(t, latestEntry.GetID(), entry.GetID())
+	})
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go:247]:
```suggestion
	})
	
```

File: internal/rsl/rsl_test.go [hunk 12]
```diff
@@ -648,6 +680,22 @@ func TestGetLatestReferenceUpdaterEntry(t *testing.T) {
...
+		}
+
+		latestEntry, err := GetLatestEntry(repo)
+		assert.Nil(t, err)
+
+		entry, annotations, err := GetLatestReferenceUpdaterEntry(repo, IsUnskipped(), ForNonGittufReference())
+		assert.Nil(t, err)
+		assert.Empty(t, annotations)
+		assert.Equal(t, latestEntry.GetID(), entry.GetID())
+	})
```
Comment (patzielinski) [inline internal/rsl/rsl_test.go:699]:
```suggestion
	})
	
```
